packages/pluggableWidgets/datagrid-date-filter-web/src/DatagridDateFilter.tsx

```diff
@@ -110,16 +112,45 @@ export default function DatagridDateFilter(props: DatagridDateFilterContainerPro
                         styles={props.style}
                         tabIndex={props.tabIndex}
                         calendarStartDay={firstDayOfWeek}
-                        updateFilters={(value: Date | null, type: DefaultFilterEnum): void => {
-                            if (
+                        updateFilters={(
+                            value: Date | null,
+                            rangeValues: [Date | null, Date | null] | null,
+                            type: DefaultFilterEnum
+                        ): void => {
+                            if (type === "between") {
+                                const [startDate, endDate] = rangeValues ?? [];
+                                let shouldExecuteOnChange = false;
+                                if (
+                                    (startDate &&
+                                        props.startDateAttribute?.value &&
+                                        !isEqual(props.startDateAttribute.value, startDate)) ||
+                                    (startDate || undefined) !== props.startDateAttribute?.value
+                                ) {
+                                    props.startDateAttribute?.setValue(startDate ?? undefined);
+                                    shouldExecuteOnChange = true;
+                                }
+                                if (
+                                    (endDate &&
+                                        props.endDateAttribute?.value &&
+                                        !isEqual(props.endDateAttribute.value, endDate)) ||
+                                    (endDate || undefined) !== props.endDateAttribute?.value
+                                ) {
+                                    props.endDateAttribute?.setValue(endDate ?? undefined);
+                                    shouldExecuteOnChange = true;
```

Comment:
Nit: I would just replace these with the actual `props.onChange?.execute()` or put the `let shouldExecuteOnChange` one block up and include the last `else if` case 😄  Cuz now there are kinda 2 things flows going on